### PR TITLE
Phpunit warning fixes appframework

### PR DESCRIPTION
**Pull request by rullzer**
Part 1 of the ongoing effort the get clean unit tests of >=phpunit-5.4

CC: @MorrisJobke @nickvergessen @LukasReschke

### CODE REVIEW

**Comment by icewind1991** [comment]
:+1: 

(`getMock` was deprecated in 5.4)

File: tests/lib/AppFramework/DependencyInjection/DIContainerTest.php [hunk 1]
```diff
@@ -36,10 +36,14 @@ class DIContainerTest extends \Test\TestCase {
...
-		$this->container = $this->getMock('OC\AppFramework\DependencyInjection\DIContainer',
-				['isAdminUser'], ['name']
-		);
-		$this->api = $this->getMock('OC\AppFramework\Core\API', array(), array('hi'));
+		$this->container = $this->getMockBuilder('OC\AppFramework\DependencyInjection\DIContainer')
+			->setMethods(['isAdminUser'])
+			->setConstructorArgs(['name'])
+			->getMock();
+		$this->api = $this->getMockBuilder('OC\AppFramework\Core\API')
+			->setMethods([])
```
Comment (nickvergessen) [inline tests/lib/AppFramework/DependencyInjection/DIContainerTest.php]: you can just skip it  ;) it was only needed because if the parameter order

**Comment by nickvergessen** [comment]
:+1:

File: tests/lib/AppFramework/DependencyInjection/DIContainerTest.php (hunk 1 above)
Comment (rullzer) [inline tests/lib/AppFramework/DependencyInjection/DIContainerTest.php]: yep very true

**Pull request merged**